src/Paramore.Brighter.MessagingGateway.MQTT/MQTTMessagingGatewayConfiguration.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Generic;
+
+namespace Paramore.Brighter.MessagingGateway.MQTT
+{
+    public class MQTTMessagingGatewayConfiguration : IAmGatewayConfiguration
+    {
+        public string ClientID { get; set; }
+        public string Username { get; set; }
+        public string Password { get; set; }
+        public bool CleanSession { get; set; }
+        public string Hostname { get; set; }
+        public object TopicPrefix { get; set; }
+        public int ConnectionAttempts { get; internal set; } = 1;
+    }
+
+
+    public class MQTTMessagingGatewayProducerConfiguration : MQTTMessagingGatewayConfiguration
+    {
+
+    }
+    public class MQTTMessagingGatewayConsumerConfiguration : MQTTMessagingGatewayConfiguration
+    {
+        public Queue<Message> queue { get; set; }
```

Comment:
I'm not sure about passing the queue in on the config. It feels as though the interface is not clean as a result. Is this for testing?

Reply:
This was added to be able to pass in an external Queue. I have removed it as it can be added later if needed.

Reply:
Let's think about our alternatives here, as it's a "surprise" coming from other transports.

Reply:
Not a blocker to going forward though.